### PR TITLE
fix(firebase_ui_localizations): Update Japanese translations

### PR DESCRIPTION
**Pull request by KoheiKanagu**
## Description

Update Japanese translation

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for _all_ changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [x] All unit tests pass (`melos run test:unit:all` doesn't fail).
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [x] No, this is _not_ a breaking change.

[issue database]: https://github.com/firebase/FirebaseUI-Flutter/issues
[Contributor Guide]: https://github.com/firebase/FirebaseUI-Flutter/blob/main/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by lesnitsky** [comment]
@kahdojay could you please find some other dev who speaks Japanese to proof-read this PR and approve?

File: packages/firebase_ui_localizations/lib/l10n/firebase_ui_ja.arb [hunk 1]
```diff
@@ -474,57 +474,57 @@
...
     "placeholders": {}
   },
-  "confirmDeleteAccountAlertMessage": "Are you sure you want to delete your account?",
+  "confirmDeleteAccountAlertMessage": "アカウントを削除してもよろしいですか？",
   "@confirmDeleteAccountAlertMessage": {
     "description": "Delete account confirmation dialog message",
     "placeholders": {}
   },
-  "confirmDeleteAccountButtonLabel": "Yes, delete",
+  "confirmDeleteAccountButtonLabel": "はい、削除する",
```
Comment (keigomichi) [inline packages/firebase_ui_localizations/lib/l10n/firebase_ui_ja.arb:527]: In my opinion, "はい、削除する" is unnatural in Japanese.

How would you like "削除する" or "はい、削除します" instead?

**Review by keigomichi — COMMENTED** [review]
@KoheiKanagu @lesnitsky 
Sorry to chime in but I'm a native Japanese speaker who was just going to submit a PR of the same content.

I have checked all diffs and made one trivial comment below.

File: packages/firebase_ui_localizations/lib/l10n/firebase_ui_ja.arb (hunk 1 above)
Comment (KoheiKanagu) [reply packages/firebase_ui_localizations/lib/l10n/firebase_ui_ja.arb:527]: @keigomichi Thanks.
That would be better.

Comment (keigomichi) [reply packages/firebase_ui_localizations/lib/l10n/firebase_ui_ja.arb]: Thanks a bunch!

**Review by keigomichi — APPROVED** [review]

**Review by russellwheatley — APPROVED** [review]

**Comment by russellwheatley** [comment]
@KoheiKanagu - do you mind merging with latest on main branch, please?

**Pull request merged**